### ISSUE TITLE
PactDslArrayJson - Best way to validate elements inside array with inconsistent order

### ISSUE THREAD

**Issue by pedrro** [DiUS/pact-jvm#817]
I have this array structure:
```
{
Array1 : ["A","B","C"],
Array2: ["D","E","F"] 
}
```

And I was validated it like this:
```
array("array1")
 .string("A")
 .string("B")
 .string("C")
.closeArray()
array("array2")
 .string("D")
 .string("E")
 .string("F")
.closeArray()
```

But my problem is that  sometimes the elements inside array aren't returned by provider in the same order, so, **my question is: How is the best way to validate it?** 
Considering that it will always return an array of 4 elements but with strings value in inconsistent order.
I already tried arrayMinLike and arrayMaxLike, but not worked.

**Comment by pedrro** [comment]
I also tried on this way:
```

.minArrayLike("array1", 1, PactDslJsonRootValue.stringMatcher("A|B|C", "A"))
    .minArrayLike("array2", 1, PactDslJsonRootValue.stringMatcher("D|E|F", "D"))
```

But pact generate a contract like:
```
array1 : [A,A,A],
array2: [D,D,D]
```

Looks like the regex doesn't work..

**Comment by abubics** [comment]
There's a distinction you might be missing in that `stringMatcher` usage vs generated examples.
`["D""D","D"]` is certainly valid, by that regex. The second argument (`"D"`) to the matcher function is the example that will be used to generate the response value.
If you provide `["D","F","E"]` in the request, it will still pass :)
In short, the example and the matcher won't give you exhaustive test data.

**Comment by pedrro** [comment]
I got it fixed, adding the last version of pact (5.8.24)
